src/Uno/Prism.Uno/Services/Dialogs/IContentDialog.cs

```diff
@@ -0,0 +1,27 @@
+using System;
+using Windows.Foundation;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+
+namespace Prism.Services.Dialogs
+{
+    public interface IContentDialog
```

Comment:
Upon closer look here, this really should be called IDialogWindow the same as we call it on WPF. Calling it IContentDialog would lead to confusion

Reply:
I understand the alignment, though in the context of UWP/Uno, a Window is a different concept (which may also be supported in the future in Uno). WPF maps to the `DialogWindow`, for UWP it maps directly to `ContentDialog`.

Reply:
Actually, it maps to the WPF Window.  The concept is more important than the underlying object type. As far as Prism is concerned, IDiaogWindow is simply the host of the dialog. Is does not have any Window related APIs associated with it.

Reply:
I would add, IContentDialog is too confusing as it implies it is the actual dialog, which it is not.

Reply:
I see your point. I did not see that `DialogWindow` was specific to Prism, I got confused by this one:

https://github.com/PrismLibrary/Prism/blob/3ab9483145e383d85f69a8660835de4a7da80cdf/src/Wpf/Prism.Wpf/Services/Dialogs/DialogWindow.xaml.cs#L8

It would make sense to make it base class in Prism.Uno that implements the `IDialogWindow`,  since developers would have to this piece of code every time the want to create a dialog:

https://github.com/unoplatform/Uno.Prism/blob/e58a476ee54fd034d5f4701e476276ae6181d55c/e2e/Uno/HelloUnoWorld.Shared/Dialogs/CustomContentDialog.xaml.cs#L20-L36

Reply:
Yes, Prism for WPF provides an IDialogWindow implementation out of the box (DialogWindow). Devs can provide a custom implementation if they like.